r1cs/gadgets/crypto/src/commitment/pedersen/mod.rs

```
@@ -49,7 +49,10 @@ where
         input: &[UInt8],
         r: &Self::RandomnessGadget,
     ) -> Result<Self::OutputGadget, SynthesisError> {
-        assert!((input.len() * 8) <= (W::WINDOW_SIZE * W::NUM_WINDOWS));
+        // TODO: check error message
+        if (input.len() * 8) > (W::WINDOW_SIZE * W::NUM_WINDOWS) {
+            return Err(SynthesisError::Other("input length verification failed".to_owned()));
```

Let's return the same error message we returned for the primitive